### ISSUE TITLE
Add list of charms in config tab for bundles

### ISSUE THREAD

**Issue by hatched** [canonical-web-and-design/charmhub.io#1105]
[Design here](https://app.zeplin.io/project/5f92a9530029eeac84ae9589/screen/5fc0ced675495db3bbef5b97)

Original bug from Jeff:
If you visit https://charmhub.io/charmed-kubernetes
You'll note that you cannot see what's in the bundle and there is no way to access the bundle yaml file to audit.

**Comment by hatched** [comment]
https://api.charmhub.io/v2/charms/info/charmed-kubernetes?fields=default-release.revision.bundle-yaml 
(add default-release.revision.bundle-yaml to the list of fields you're retrieving). It'll be the content of bundle.yaml, verbatim.

**Comment by anthonydillon** [comment]
I think we need to provide a method of downloading the contents of a charm or bundle. I see users using Juju this way, a lot. There are a few reasons:
1. To learn Juju. When writing a charm it's often key to look at how other similar charms have been written and copy aspects of them.
2. To see what they are doing. To truly understand the charm. Maybe security auditing, although I don't see that so much in development.
3. To extend a bundle. Bundles are just a Yaml file of charms, units and relations. Often you want to use a bundle but slightly extend it for your use case. Like making the openstack bundle HA. 
4. For usage in closed and seemingly local systems. Often environments will be behind firewalls and you just want to download, scp and side load the bundle which keeps a local library of bundle files.

**Comment by danielmutis** [comment]
Also, for context, the original designs for the bundle pages included an image of the topology ([view design](https://zpl.io/bArP4JP)) and a configuration tab with all the charms inside the bundle and config details for each charm ([view design](https://zpl.io/brDRNp1)).

**Comment by niemeyer** [comment]
We can certainly discuss a way to visualize content in a nice way.  I did express concerns in our conversations on the topic, and was the one strongly advocating for removal of what used to be there before.  The issue isn't the visualization itself, though, but the fact that there were a number of problems for which the answer was digging into the content and opening it up to understand the charm.  We don't want _that_, specifically.  If you're trying to know which actions exist, there should be documentation for that which is not opening it up and reading code. If you want to know which endpoints exist, again there should be support for that in our interface, instead of expecting people to open up a yaml file. And so forth.

Also, a number of the use cases listed above will fall short.  Copying an individual charm doesn't copy its resources, or images, or any other server side metadata that is on the store.  Designing our workflows as "just copy a blob and push it as your own" is again, a workaround to story that we didn't really think through.

So, in summary, it's not about preventing people from visualizing the content, but about not designing workflows in what's supposed to be a beautiful user interface which are essentially telling people to read code and copy blobs around.

**Comment by hatched** [comment]
When learning how to do something I find it invaluable to be able to look at existing code to see how others have solved various problems. Not being able to download a charm and bundle for auditing is a huge impediment for me. I'd also expect anyone wanting to run workloads in production would also want to audit what code is actually being executed in a charm before they hit deploy.

@danielmutis bundles contain a lot more data than just an application list and configuration changes. We can implement a pretty UI for every possible element in the bundle yaml but until that's done, providing a link to the asset is a trivial first step...and I'd argue should remain indefinitely for those wanting the actual asset.

I created this issue because someone was reporting that a bundle was not being deployed with the correct annotations but there was no way to audit what was in the bundle before deploying so I had to go to jaas.ai and download the yaml from there and deploy it manually to investigate. I was able to get assistance on the correct API path to download the asset as mentioned above but we shouldn't expect users to have to dig through the API for this information.

**Comment by jkfran** [comment]
We implemented already the Configure tab for bundles. There are some improvements coming with https://github.com/canonical-web-and-design/charmhub.io/issues/1163
